index.js

```diff
@@ -37,15 +38,29 @@ module.exports = function(userOpts) {
 		.readdirSync(opts.dir)
 		.forEach(function(filename) {
 			var file = path.join(opts.dir, filename);
+			var extension = path.extname(filename);
 			var stat = fs.statSync(file);
 
-			if (stat.isFile() && opts.extensions.indexOf(path.extname(filename)) == -1) {
+			if (stat.isFile() && opts.extensions.indexOf(extension) === -1) {
 				return;
 			}
 
-			var taskname = path.basename(filename, path.extname(filename));
+			var taskname = path.basename(filename, extension);
 			var taskinfo = require(file);
 
+			if (opts.params !== undefined) {
+				if (!Array.isArray(taskInfo)) {
```

Comment:
Probably want to `%s/taskInfo/taskinfo/g` on this.  That's the only issue I'm seeing.